### PR TITLE
Task/apps 3244 update dev dependencies

### PR DESCRIPTION
**Pull request by solomontang**
- bumped babel and webpack to latest minor
  - remove plugins that are included by default in `babel/preset-env`
  - https://babel.dev/blog/2021/04/29/7.14.0#new-class-features-enabled-by-default
- jest updated to v27
- stylelint updated to v14
- eslint updated to v7.32
- eslint and flow configs adjusted to allow new jsx transform
  - ran codemod to remove unnecessary react imports

### CODE REVIEW

**Review by anzioka — APPROVED** [review]

File: config/babel/babel.config.js [hunk 1]
```diff
@@ -34,7 +32,7 @@ module.exports = {
   ],
   presets: [
     ['@babel/preset-env', {
-      corejs: '3.12',
+      corejs: '3.19',
```
Comment (UnsungHero97) [inline config/babel/babel.config.js:35]: It shouldn't be a problem, but dependencies are not yet on 3.19; let's keep an eye on it on staging.

File: src/containers/study/components/DeleteParticipantModal.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
 // @flow
 
-import React from 'react';
-
 import {
-  // $FlowFixMe
+// $FlowFixMe
```
Comment (UnsungHero97) [inline src/containers/study/components/DeleteParticipantModal.js:4]: No change required. This shouldn't be necessary. I've been able to remove this in other apps.

File: src/index.js [hunk 3]
```diff
@@ -47,15 +47,15 @@ const NormalizeCSS = createGlobalStyle`
 const GlobalStyle = createGlobalStyle`
   @supports (font-variation-settings: normal) {
     html {
-      font-family: 'Inter var', sans-serif;
+      font-family: "Inter var", sans-serif;
```
Comment (UnsungHero97) [inline src/index.js:50]: Why do you need quotes here but not on line 58?

**Review by UnsungHero97 — APPROVED** [review]

File: src/index.js (hunk 3 above)
Comment (solomontang) [reply src/index.js:50]: Eslint rules for quotes on font families changed

File: config/babel/babel.config.js (hunk 1 above)
Comment (solomontang) [reply config/babel/babel.config.js:35]: Corejs has been updated in this pr to 3.19

Comment (UnsungHero97) [reply config/babel/babel.config.js:35]: I'm referring to `lattice`, `lattice-sagas`, `lattice-utils`, etc.

File: src/index.js (hunk 3 above)
Comment (UnsungHero97) [reply src/index.js:50]: Why is it not consistent though?

Comment (solomontang) [reply src/index.js:50]: https://stylelint.io/user-guide/rules/list/font-family-name-quotes

**Pull request merged**